### ISSUE TITLE
Outlook addin is not able to get updated context while pinned

### ISSUE THREAD

**Issue by somenathmaji** [OfficeDev/office-js#2076]
## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: Microsoft 365 Build >= 16.0.14228
* Operating System: Windows 10
* Browser (if using Office on the web): N/A
 
## Expected behavior
When we change mail the `Office.context.mailbox.item` should be picked from the current mail.

## Current behavior
When we change the mail the `Office.context.mailbox.item` is not updating based on the currently selected mail.

## Steps to reproduce
We are not able to reproduce it in ScriptLab. The below steps are to reproduce this issue in our addin.
1. Open a mail
2. Open the addin and pin it
3. Go to some other mail
4. Check in debug console `Office.context.mailbox.item` is not getting updated based on currently selected mail.

## Link to live example(s)
Below is the [link](https://celigodev-my.sharepoint.com/:v:/g/personal/somenath_cloudextend_dev/ETd7Aoi3D4dIiQprI5SAivsBtWggFbxaLTX-EdK9ZjhsDA) of the video from the addin -
1. In the video we followed the above steps
2. We tried to get `Office.context.mailbox.item.subject` on a different mail
3. It didn't show information from the current context

## Context
- We have one contextual outlook addin. Lots of customers are facing this issue with recent outlook update and are not able to use the addin with pinning. 
- As of now, in all the cases the addin is using Edge HTML. We tried to install WebView2 and it fixed the issue for few customers. While for others the addin is still using Edge HTML.
- Their OS and Office versions meet all the criteria to use WebView2 from this [link](https://docs.microsoft.com/en-us/office/dev/add-ins/concepts/browsers-used-by-office-web-add-ins).

Apart from the original issue -
Can we also get some insight into which scenarios the addin could still keep using Edge HTML?

**Comment by KevlarBat** [comment]
Same issue : Our customers using old Edge web view are not enable to update their add-in anymore. 
I delete the webview2 on my personal computer to test it and I have the same problem. 
As a clue, I figure out that this issue only append when trying to subscribe to the ItemChanged event inside the Angular context of our application (or any other PoC with Angular). I try to subscribe to the event at the root Ts file just after office initialization and it worked. I do not know about other frameworks. 
Hope this can help.

**Comment by exextoc** [comment]
@somenathmaji based on the response from @KevlarBat, are you using Angular?
If yes, does @KevlarBat's response help?

**Comment by KevlarBat** [comment]
There is a misunderstanding : 
The change events from the office API are not trigger at all as soon as I initialize my Angular app.
this work : 
```
   // Set up ItemChanged event
    Office.context.mailbox.addHandlerAsync(Office.EventType.ItemChanged, selectedMailItemChanged);

    /* launch Angular app
    platformBrowserDynamic().bootstrapModule(AppModule)
        .catch(err => console.error(err));
     */
```
this do not : 
```
    // Set up ItemChanged event
    Office.context.mailbox.addHandlerAsync(Office.EventType.ItemChanged, selectedMailItemChanged);

    // launch Angular app
    platformBrowserDynamic().bootstrapModule(AppModule)
        .catch(err => console.error(err));
```
        
The issue is not resolve and I still need help

**Comment by somenathmaji** [comment]
@exextoc Yes, we are also using Angular. @KevlarBat Are you facing this issue on a specific outlook build? We noticed this issue first in outlook build 14228 and it's still present in the latest build 14326.

**Comment by KevlarBat** [comment]
@somenathmaji I am currently on build 14326 and we also noticed it in 14228.

**Comment by somenathmaji** [comment]
@exextoc Do we have any updates?

We also want to know how could the app keep using Edge HTML in spite of webview2 being installed? (the OS and Office meets the required versions)

Any help is appreciated.

**Comment by exextoc** [comment]
To implement a pinnable taskpane, make sure that you have added the SupportsPinning element to the Action element in the manifest that describes the task pane button. The SupportsPinning element is defined in the VersionOverrides v1.1 schema, so you will need to include a VersionOverrides element both for v1.0 and v1.1.

Full instructions for implementing the pinnable taskpane can be found [here](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/pinnable-taskpane).

When you register the event handler, this should be done in the Office.initialize function for your task pane, as described [here](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/pinnable-taskpane#register-the-event-handler).

If you are still having trouble, we have a [demo add-in](https://docs.microsoft.com/en-us/office/dev/add-ins/outlook/pinnable-taskpane#see-also) for you to try. If this add-in does not work correctly, please let us know.

Here's some information about the runtimes :
https://developer.microsoft.com/en-us/office/blogs/understanding-office-add-ins-runtime/
As mentioned this doc, you could try to uninstall the WebView2 control to use the older webview control (EdgeHTML).

**Comment by somenathmaji** [comment]
We are able to reproduce this issue in a test angular addin. Here is the link to the test addin [repo](https://github.com/somenathmaji/outlook-test-addin/tree/main)
It's reproducible when we are navigating to another view inside the ItemChanged event handler.
Shall we re-open this ticket?

**Comment by AlexJerabek** [comment]
Thank you for the test repo and repro steps @somenathmaji. 

I'll reopen this issue and ping @exextoc.

**Comment by exextoc** [comment]
If you navigate to a different webpage within the taskpane, you need to ensure that the new webpage has OfficeJS and also re-registers for events so that we get the new callbacks. Otherwise, events won't get to the add-in. So the new webpage should also have office.initialize.

**Comment by somenathmaji** [comment]
@exextoc We are not navigating to a different webpage. It's an Angular app and we are navigating to a different view.
The `ItemChanged` events handler used to work with Edge HTML and it's working in WebView2.

**Comment by exextoc** [comment]
Based on your past responses, can you try to debug the Angular app following the documentation here: https://docs.microsoft.com/en-us/office/dev/add-ins/testing/debug-add-ins-using-f12-developer-tools-on-windows

You also mentioned that you could not repro with ScriptLab so this might have something to do with the Angular add-in itself, were there any major changes in the add-in that happened when it stopped working with Edge?

**Comment by somenathmaji** [comment]
@exextoc We didn't release any updates during the time when we got this issue. There was only an Outlook update. This is only happening with Edge HTML runtime after the Outlook update. We found a workaround to use WebView2 from this [documentation](https://developer.microsoft.com/en-us/office/blogs/understanding-office-add-ins-runtime/) -
>Add registry key HKEY_CURRENT_USER\SOFTWARE\Microsoft\Office\16.0\WEF\Win32WebView2 and set its value to dword:00000001

This seems to be working for the Semi-Annual Enterprise channel and Current channel users.
